### ISSUE TITLE
💅 `useArrayLiterals` safe fix rewrites `Array(...args)` to `[...args]`

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11942]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.14

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:           npm
  JS_RUNTIME_NAME:              node
```

</details>

### Rule name

`style/useArrayLiterals`

### Playground link

https://biomejs.dev/playground/?lintRules=useArrayLiterals&fix=safeFixes#files.useArrayLiterals.js=YwBvAG4AcwB0ACAAYQByAGcAcwAgAD0AIABbADMAXQA7AAoAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAQQByAHIAYQB5ACgALgAuAC4AYQByAGcAcwApAC4AbABlAG4AZwB0AGgALAAgAG4AZQB3ACAAQQByAHIAYQB5ACgALgAuAC4AYQByAGcAcwApAC4AbABlAG4AZwB0AGgAKQA7AAoA

```js
const args = [3];
console.log(Array(...args).length, new Array(...args).length);
```

`Array(...args)` with `args = [3]` is `Array(3)`, an array of length 3. The safe fix produces `[...args]`, an array of length 1 containing `3`:

| | output |
| --- | --- |
| before fix | `3 3` |
| after fix | `1 1` |

With a spread argument the rule cannot know how many arguments there are, or whether a single one is a number.

Reproduced with the 2.5.14 CLI (`biome lint --write`) and in the playground (f74eb38).

### Expected result

`Array(...x)` and `new Array(...x)` are not reported, or at least not fixed, since the argument count is unknown.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

---

Tool Use Disclaimer: Claude Code using Anthropic Opus 5.5 helped build the reproduction.
